map_view/set-up-location-driven-geotriggers/src/main/java/com/esri/samples/set_up_location_driven_geotriggers/SetUpLocationDrivenGeotriggersController.java

```diff
@@ -0,0 +1,305 @@
+            featureFenceParametersGardenSection, arcadeExpression, "Section Geotrigger");
+          FenceGeotrigger fenceGeotriggerPOI = new FenceGeotrigger(locationGeotriggerFeed, FenceRuleType.ENTER_OR_EXIT,
+            featureFenceParametersPOI, arcadeExpression, "POI Geotrigger");
+
+          // create geotrigger monitors from the fence geotriggers
+          gardenSectionGeotriggerMonitor = new GeotriggerMonitor(fenceGeotriggerGardenSection);
+          gardenPOIGeotriggerMonitor = new GeotriggerMonitor(fenceGeotriggerPOI);
+          List<GeotriggerMonitor> geotriggerMonitors = new ArrayList<>(Arrays.asList(
+            gardenSectionGeotriggerMonitor, gardenPOIGeotriggerMonitor));
+
+          // for each geotrigger monitor, add a notification event listener, and start the monitor
+          geotriggerMonitors.forEach(monitor -> {
+
+            monitor.addGeotriggerMonitorNotificationEventListener(notificationEvent -> {
+              // fence geotrigger notification info provides access to the feature that triggered the notification
+              var fenceGeotriggerNotificationInfo = (FenceGeotriggerNotificationInfo) notificationEvent.getGeotriggerNotificationInfo();
+
+              // get the name of the fence feature
+              fenceFeatureName = fenceGeotriggerNotificationInfo.getMessage();
+              geoTriggerName = fenceGeotriggerNotificationInfo.getGeotriggerMonitor().getGeotrigger().getName();
+              // determine the notification type on the notification info (entered or exited)
+              FenceNotificationType fenceNotificationType = fenceGeotriggerNotificationInfo.getFenceNotificationType();
+              // get a reference to the fence feature that triggered the notification
+              ArcGISFeature fenceFeature = (ArcGISFeature) fenceGeotriggerNotificationInfo.getFenceGeoElement();
+
+              // when entering a given geofence, add the feature's information to the UI
+              if (fenceNotificationType == FenceNotificationType.ENTERED) {
+
+                // add the description from the feature's attributes to the UI
+                handleAddingFeatureInfoToUI(fenceFeature);
+
+              } else if (fenceNotificationType == FenceNotificationType.EXITED) {
+                // when exiting a given geofence, remove its information from the UI
+                handleRemovingFeatureInfoFromUI();
+              }
+            });
+
+            // start the geotrigger monitor
+            monitor.startAsync();
+
+          });
+        } else new Alert(Alert.AlertType.ERROR, "Simulated data location source failed to start").show();
+      });
+
+    });
+  }
+
+  /**
+   * Creates and starts a simulated location data source based on a json file containing a set polyline data path.
+   */
+  private void initializeSimulatedLocationDisplay() throws IOException {
+
+    // access the json of the walking route points
```

Comment:
maybe something like: read a json string from a file which contains a set of points collected along a walking route